lib/js/protocol/enums/ControlFrameTags.js

```diff
@@ -30,87 +30,92 @@
 class ControlFrameTags {
-
-    StartServiceACKBase = class StartServiceACKBase {
-        MTU = "mtu";
-    }
-
-    NAKBase = class NAKBase {
-        REJECTED_PARAMS = "rejectedParams";
-    }
-
-    RPC = class RPC {
-        StartService = class StartService {
-            /** The max version of the protocol supported by client requesting service to start.<br>
-			* Must be in the format "Major.Minor.Patch"
-			*/
-            PROTOCOL_VERSION = "protocolVersion";
-        }
-
-        StartServiceACK = class StartServiceACK extends StartServiceACKBase {
-            /** The negotiated version of the protocol. Must be in the format "Major.Minor.Patch"*/
-            PROTOCOL_VERSION = StartService.PROTOCOL_VERSION;
-            /** Hash ID to identify this service and used when sending an EndService control frame*/
-            HASH_ID = "hashId";
-            /** HU allowed transport for secondary connection */
-            SECONDARY_TRANSPORTS = "secondaryTransports";
-            /** HU allowed transports for audio and video services (1 == Primary, 2 == Secondary) */
-            AUDIO_SERVICE_TRANSPORTS = "audioServiceTransports";
-            VIDEO_SERVICE_TRANSPORTS = "videoServiceTransports";
-            /** Auth token to be used for log in into services **/
-            AUTH_TOKEN = "authToken";
-        }
-
-        StartServiceNAK = class StartServiceNAK extends NAKBase {}
-
-        EndService = class EndService {
-            HASH_ID = RPC.StartServiceACK.HASH_ID;
-        }
-
-        EndServiceACK = class EndServiceACK {}
-
-        EndServiceNAK = class EndServiceNAK extends NAKBase {}
-
-        TransportEventUpdate = class TransportEventUpdate {
-            TCP_IP_ADDRESS = "tcpIpAddress";
-            TCP_PORT = "tcpPort";
-        }
-
-        RegisterSecondaryTransport = class RegisterSecondaryTransport {}
-
-        RegisterSecondaryTransportACK = class RegisterSecondaryTransportACK {}
-
-        RegisterSecondaryTransportNAK = class RegisterSecondaryTransportNAK extends NAKBase {
-            REASON = "reason";
-        }
-    }
-
-
-    Audio = class Audio {
-        StartService = class StartService {}
-        StartServiceACK = class StartServiceACK extends StartServiceACKBase {}
-        StartServiceNAK = class StartServiceNAK extends NAKBase {}
-        EndService = class EndService {}
-        EndServiceACK = class EndServiceACK {}
-        EndServiceNAK = class EndServiceNAK extends NAKBase {}
-    }
-
-    Video = class Video {
-        StartService = class StartService {
-            HEIGHT = "height";
-            WIDTH = "width";
-            VIDEO_PROTOCOL = "videoProtocol";
-            VIDEO_CODEC = "videoCodec";
-        }
-        StartServiceACK = class StartServiceACK extends StartServiceACKBase {
-            HEIGHT = StartService.HEIGHT;
-            WIDTH = StartService.WIDTH;
-
-        }
-    }
 }
 
+// hidden inner objects
+
+const StartServiceACKBase = {
```

Comment:
Seems that most of these classes are essentially enums. Might be an opportunity for refactoring in the future.